<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<!-- 2021-05-09 Sun 02:53 -->
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>COVID-19 PubSeq Uploading Data (part 3)</title>
<meta name="generator" content="Org mode" />
<meta name="author" content="Pjotr Prins" />
<style type="text/css">
 <!--/*--><![CDATA[/*><!--*/
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #ccc;
    box-shadow: 3px 3px 3px #eee;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: auto;
    padding-top: 1.2em;
  }
  pre.src:before {
    display: none;
    position: absolute;
    background-color: white;
    top: -10px;
    right: 10px;
    padding: 3px;
    border: 1px solid black;
  }
  pre.src:hover:before { display: inline;}
  /* Languages per Org manual */
  pre.src-asymptote:before { content: 'Asymptote'; }
  pre.src-awk:before { content: 'Awk'; }
  pre.src-C:before { content: 'C'; }
  /* pre.src-C++ doesn't work in CSS */
  pre.src-clojure:before { content: 'Clojure'; }
  pre.src-css:before { content: 'CSS'; }
  pre.src-D:before { content: 'D'; }
  pre.src-ditaa:before { content: 'ditaa'; }
  pre.src-dot:before { content: 'Graphviz'; }
  pre.src-calc:before { content: 'Emacs Calc'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-fortran:before { content: 'Fortran'; }
  pre.src-gnuplot:before { content: 'gnuplot'; }
  pre.src-haskell:before { content: 'Haskell'; }
  pre.src-hledger:before { content: 'hledger'; }
  pre.src-java:before { content: 'Java'; }
  pre.src-js:before { content: 'Javascript'; }
  pre.src-latex:before { content: 'LaTeX'; }
  pre.src-ledger:before { content: 'Ledger'; }
  pre.src-lisp:before { content: 'Lisp'; }
  pre.src-lilypond:before { content: 'Lilypond'; }
  pre.src-lua:before { content: 'Lua'; }
  pre.src-matlab:before { content: 'MATLAB'; }
  pre.src-mscgen:before { content: 'Mscgen'; }
  pre.src-ocaml:before { content: 'Objective Caml'; }
  pre.src-octave:before { content: 'Octave'; }
  pre.src-org:before { content: 'Org mode'; }
  pre.src-oz:before { content: 'OZ'; }
  pre.src-plantuml:before { content: 'Plantuml'; }
  pre.src-processing:before { content: 'Processing.js'; }
  pre.src-python:before { content: 'Python'; }
  pre.src-R:before { content: 'R'; }
  pre.src-ruby:before { content: 'Ruby'; }
  pre.src-sass:before { content: 'Sass'; }
  pre.src-scheme:before { content: 'Scheme'; }
  pre.src-screen:before { content: 'Gnu Screen'; }
  pre.src-sed:before { content: 'Sed'; }
  pre.src-sh:before { content: 'shell'; }
  pre.src-sql:before { content: 'SQL'; }
  pre.src-sqlite:before { content: 'SQLite'; }
  /* additional languages in org.el's org-babel-load-languages alist */
  pre.src-forth:before { content: 'Forth'; }
  pre.src-io:before { content: 'IO'; }
  pre.src-J:before { content: 'J'; }
  pre.src-makefile:before { content: 'Makefile'; }
  pre.src-maxima:before { content: 'Maxima'; }
  pre.src-perl:before { content: 'Perl'; }
  pre.src-picolisp:before { content: 'Pico Lisp'; }
  pre.src-scala:before { content: 'Scala'; }
  pre.src-shell:before { content: 'Shell Script'; }
  pre.src-ebnf2ps:before { content: 'ebfn2ps'; }
  /* additional language identifiers per "defun org-babel-execute"
       in ob-*.el */
  pre.src-cpp:before  { content: 'C++'; }
  pre.src-abc:before  { content: 'ABC'; }
  pre.src-coq:before  { content: 'Coq'; }
  pre.src-groovy:before  { content: 'Groovy'; }
  /* additional language identifiers from org-babel-shell-names in
     ob-shell.el: ob-shell is the only babel language using a lambda to put
     the execution function name together. */
  pre.src-bash:before  { content: 'bash'; }
  pre.src-csh:before  { content: 'csh'; }
  pre.src-ash:before  { content: 'ash'; }
  pre.src-dash:before  { content: 'dash'; }
  pre.src-ksh:before  { content: 'ksh'; }
  pre.src-mksh:before  { content: 'mksh'; }
  pre.src-posh:before  { content: 'posh'; }
  /* Additional Emacs modes also supported by the LaTeX listings package */
  pre.src-ada:before { content: 'Ada'; }
  pre.src-asm:before { content: 'Assembler'; }
  pre.src-caml:before { content: 'Caml'; }
  pre.src-delphi:before { content: 'Delphi'; }
  pre.src-html:before { content: 'HTML'; }
  pre.src-idl:before { content: 'IDL'; }
  pre.src-mercury:before { content: 'Mercury'; }
  pre.src-metapost:before { content: 'MetaPost'; }
  pre.src-modula-2:before { content: 'Modula-2'; }
  pre.src-pascal:before { content: 'Pascal'; }
  pre.src-ps:before { content: 'PostScript'; }
  pre.src-prolog:before { content: 'Prolog'; }
  pre.src-simula:before { content: 'Simula'; }
  pre.src-tcl:before { content: 'tcl'; }
  pre.src-tex:before { content: 'TeX'; }
  pre.src-plain-tex:before { content: 'Plain TeX'; }
  pre.src-verilog:before { content: 'Verilog'; }
  pre.src-vhdl:before { content: 'VHDL'; }
  pre.src-xml:before { content: 'XML'; }
  pre.src-nxml:before { content: 'XML'; }
  /* add a generic configuration mode; LaTeX export needs an additional
     (add-to-list 'org-latex-listings-langs '(conf " ")) in .emacs */
  pre.src-conf:before { content: 'Configuration File'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .equation-container {
    display: table;
    text-align: center;
    width: 100%;
  }
  .equation {
    vertical-align: middle;
  }
  .equation-label {
    display: table-cell;
    text-align: right;
    vertical-align: middle;
  }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  .org-svg { width: 90%; }
  /*]]>*/-->
</style>
<link rel="Blog stylesheet" type="text/css" href="blog.css" />
<script type="text/javascript">
// @license magnet:?xt=urn:btih:e95b018ef3580986a04669f1b5879592219e2a7a&dn=public-domain.txt Public Domain
<!--/*--><![CDATA[/*><!--*/
     function CodeHighlightOn(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.add("code-highlighted");
         target.classList.add("code-highlighted");
       }
     }
     function CodeHighlightOff(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.remove("code-highlighted");
         target.classList.remove("code-highlighted");
       }
     }
    /*]]>*///-->
// @license-end
</script>
</head>
<body>
<div id="content">
<h1 class="title">COVID-19 PubSeq Uploading Data (part 3)</h1>
<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#orgdce1d82">1. Introduction</a></li>
<li><a href="#org0b76809">2. Uploading data</a></li>
<li><a href="#orga924255">3. Step 1: Upload sequence</a></li>
<li><a href="#org5f791e4">4. Step 2: Add metadata</a>
<ul>
<li><a href="#org70a4825">4.1. Obligatory fields</a>
<ul>
<li><a href="#orgd7dd05b">4.1.1. Sample ID (sample_id)</a></li>
<li><a href="#orgbfa5e6f">4.1.2. Collection date</a></li>
<li><a href="#org9a767b6">4.1.3. Collection location</a></li>
<li><a href="#org78d41f5">4.1.4. Sequencing technology</a></li>
<li><a href="#org8a91e82">4.1.5. Authors</a></li>
</ul>
</li>
<li><a href="#org1b9ae44">4.2. Optional fields</a>
<ul>
<li><a href="#org5c6db3e">4.2.1. Host information</a></li>
<li><a href="#orgd6880ca">4.2.2. Collecting institution</a></li>
<li><a href="#org53c6a46">4.2.3. Specimen source</a></li>
<li><a href="#org34a9c9a">4.2.4. Source database accession</a></li>
<li><a href="#orgb572a75">4.2.5. Strain name</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#org53381c7">5. Step 3: Submit to COVID-19 PubSeq</a>
<ul>
<li><a href="#org5d317ac">5.1. Trouble shooting</a></li>
</ul>
</li>
<li><a href="#org389fedb">6. Step 4: Check output</a></li>
<li><a href="#org3fb7fc8">7. Bulk sequence uploader</a>
<ul>
<li><a href="#org3514567">7.1. Run the uploader (CLI)</a></li>
<li><a href="#orgc97a930">7.2. Example: uploading bulk GenBank sequences</a>
<ul>
<li><a href="#orga76159c">7.2.1. List PubSeq IDs</a></li>
<li><a href="#org89757c4">7.2.2. Fetch GenBank IDs</a></li>
<li><a href="#org538bf18">7.2.3. Fetch GenBank XML</a></li>
<li><a href="#orgf96d0f6">7.2.4. Transform and normalize data for uploading</a></li>
<li><a href="#org274ef05">7.2.5. Normalize metadata</a></li>
<li><a href="#org2351e31">7.2.6. Upload data</a></li>
</ul>
</li>
<li><a href="#org5985f41">7.3. Example: preparing metadata from spreadsheets</a></li>
</ul>
</li>
</ul>
</div>
</div>

<div id="outline-container-orgdce1d82" class="outline-2">
<h2 id="orgdce1d82"><span class="section-number-2">1</span> Introduction</h2>
<div class="outline-text-2" id="text-1">
<p>
In this document we explain how to upload data into COVID-19 PubSeq.
This can happen through a web page, or through a command line
script. We'll also show how to parametrize uploads by using templates.
The procedure is much easier than with other sequence repositories and
PubSeq uploads can be fully automated. Once uploaded you can use our
export API to prepare for other repositories, including GenBank.
</p>
</div>
</div>

<div id="outline-container-org0b76809" class="outline-2">
<h2 id="org0b76809"><span class="section-number-2">2</span> Uploading data</h2>
<div class="outline-text-2" id="text-2">
<p>
PubSeq allows you to upload your SARS-Cov-2 strains to a public
resource for global comparisons. A recompute of the pangenome gets
triggered on upload. Read the <a href="./about">ABOUT</a> page for more information.
</p>
</div>
</div>

<div id="outline-container-orga924255" class="outline-2">
<h2 id="orga924255"><span class="section-number-2">3</span> Step 1: Upload sequence</h2>
<div class="outline-text-2" id="text-3">
<p>
To upload a sequence with the <a href="http://covid19.genenetwork.org/">web upload page</a> hit the File browse
button and select the FASTA file on your local hard disk.
</p>

<p>
We start with an assembled or mapped sequence in FASTA format. The
PubSeq uploader contains a <a href="https://github.com/arvados/bh20-seq-resource/blob/master/bh20sequploader/qc_fasta.py">QC step</a> which checks whether it is a likely
SARS-CoV-2 sequence. While PubSeq deduplicates sequences and never
overwrites metadata, you may still want to check whether your data
already is in the system by querying some metadata as described in
<a href="./blog?id=using-covid-19-pubseq-part1">Query metadata with SPARQL</a> or by simply downloading and checking one
of the files on the <a href="./download">download</a> page. We find GenBank <a href="https://www.ncbi.nlm.nih.gov/nuccore/MT536190">MT536190.1</a> has not
been included yet. A FASTA text file can be <a href="https://www.ncbi.nlm.nih.gov/nuccore/MT536190.1?report=fasta&amp;log$=seqview&amp;format=text">downloaded</a> to your local
disk and uploaded through our <a href="./">web upload page</a>. Make sure the file is
correct FASTA and does not include any HTML!
</p>

<p>
Note: we currently only allow FASTA uploads. In the near future we'll
allow for uploading raw sequence files. This is important for creating
an improved pangenome.
</p>
</div>
</div>

<div id="outline-container-org5f791e4" class="outline-2">
<h2 id="org5f791e4"><span class="section-number-2">4</span> Step 2: Add metadata</h2>
<div class="outline-text-2" id="text-4">
<p>
The <a href="./">web upload page</a> contains fields for adding metadata. Metadata is
important for analysis. The metadata is available for queries, see
<a href="./blog?id=using-covid-19-pubseq-part1">Query metadata with SPARQL</a>, and can be used to annotate variations of
the virus in different ways. Metadata also includes attribution
details.
</p>

<p>
A number of fields are obligatory: sample id, date, location,
technology and authors. The others are optional, but it is valuable to
enter them when information is available. Metadata is defined in this
<a href="https://github.com/arvados/bh20-seq-resource/blob/master/bh20sequploader/bh20seq-schema.yml">schema</a>. From this schema we generate the input form. Note that
optional fields have a question mark in the <code>type</code>. You can add code
for metadata yourself because this is a public resource! See also
<a href="./blog?id=using-covid-19-pubseq-part5">Modify metadata</a> for more information. The fields are:
</p>
</div>

<div id="outline-container-org70a4825" class="outline-3">
<h3 id="org70a4825"><span class="section-number-3">4.1</span> Obligatory fields</h3>
<div class="outline-text-3" id="text-4-1">
</div>
<div id="outline-container-orgd7dd05b" class="outline-4">
<h4 id="orgd7dd05b"><span class="section-number-4">4.1.1</span> Sample ID (sample_id)</h4>
<div class="outline-text-4" id="text-4-1-1">
<p>
This is a string field that defines a unique sample identifier by the
submitter. In addition to sample_id we also have host_id,
provider and submitter_sample_id where host is the host the
sample came from, provider sample is the institution id and
submitter is the submitting individual id. host_id is important when
multiple sequences come from the same host. Make sure not to have
spaces in the sample_id.
</p>

<p>
Here we add the GenBank ID MT536190.1.
</p>
</div>
</div>

<div id="outline-container-orgbfa5e6f" class="outline-4">
<h4 id="orgbfa5e6f"><span class="section-number-4">4.1.2</span> Collection date</h4>
<div class="outline-text-4" id="text-4-1-2">
<p>
Estimated collection date. The GenBank page says April 6, 2020.
</p>
</div>
</div>

<div id="outline-container-org9a767b6" class="outline-4">
<h4 id="org9a767b6"><span class="section-number-4">4.1.3</span> Collection location</h4>
<div class="outline-text-4" id="text-4-1-3">
<p>
A search on wikidata says Los Angeles is
<a href="https://www.wikidata.org/entity/Q65">https://www.wikidata.org/entity/Q65</a>
</p>
</div>
</div>

<div id="outline-container-org78d41f5" class="outline-4">
<h4 id="org78d41f5"><span class="section-number-4">4.1.4</span> Sequencing technology</h4>
<div class="outline-text-4" id="text-4-1-4">
<p>
GenBank entry says Illumina, so we can fill that in
</p>
</div>
</div>

<div id="outline-container-org8a91e82" class="outline-4">
<h4 id="org8a91e82"><span class="section-number-4">4.1.5</span> Authors</h4>
<div class="outline-text-4" id="text-4-1-5">
<p>
GenBank entry says 'Lamers,S., Nolan,D.J., Rose,R., Cross,S., Moraga
Amador,D., Yang,T., Caruso,L., Navia,W., Von Borstel,L., Hui Zhou,X.,
Freehan,A. and Garcia-Diaz,J.', so we can fill that in.
</p>
</div>
</div>
</div>

<div id="outline-container-org1b9ae44" class="outline-3">
<h3 id="org1b9ae44"><span class="section-number-3">4.2</span> Optional fields</h3>
<div class="outline-text-3" id="text-4-2">
<p>
All other fields are optional. But let's see what we can add.
</p>
</div>

<div id="outline-container-org5c6db3e" class="outline-4">
<h4 id="org5c6db3e"><span class="section-number-4">4.2.1</span> Host information</h4>
<div class="outline-text-4" id="text-4-2-1">
<p>
Sadly, not much is known about the host from GenBank. A little
sleuthing renders an interesting paper by some of the authors titled
<a href="https://www.medrxiv.org/content/10.1101/2020.04.24.20078691v1">SARS-CoV-2 is consistent across multiple samples and methodologies</a>
which dates after the sample, but has no reference other than that the
raw data came from the SRA database, so it probably does not describe
this particular sample. We don't know what this strain of SARS-Cov-2
did to the person and what the person was like (say age group).
</p>
</div>
</div>

<div id="outline-container-orgd6880ca" class="outline-4">
<h4 id="orgd6880ca"><span class="section-number-4">4.2.2</span> Collecting institution</h4>
<div class="outline-text-4" id="text-4-2-2">
<p>
We can fill that in.
</p>
</div>
</div>

<div id="outline-container-org53c6a46" class="outline-4">
<h4 id="org53c6a46"><span class="section-number-4">4.2.3</span> Specimen source</h4>
<div class="outline-text-4" id="text-4-2-3">
<p>
We have that: nasopharyngeal swab
</p>
</div>
</div>

<div id="outline-container-org34a9c9a" class="outline-4">
<h4 id="org34a9c9a"><span class="section-number-4">4.2.4</span> Source database accession</h4>
<div class="outline-text-4" id="text-4-2-4">
<p>
Genbank which is <a href="http://identifiers.org/insdc/MT536190.1#sequence">http://identifiers.org/insdc/MT536190.1#sequence</a>.
Note we plug in our own identifier MT536190.1.
</p>
</div>
</div>

<div id="outline-container-orgb572a75" class="outline-4">
<h4 id="orgb572a75"><span class="section-number-4">4.2.5</span> Strain name</h4>
<div class="outline-text-4" id="text-4-2-5">
<p>
SARS-CoV-2/human/USA/LA-BIE-070/2020
</p>
</div>
</div>
</div>
</div>

<div id="outline-container-org53381c7" class="outline-2">
<h2 id="org53381c7"><span class="section-number-2">5</span> Step 3: Submit to COVID-19 PubSeq</h2>
<div class="outline-text-2" id="text-5">
<p>
Once you have the sequence and the metadata together, hit
the 'Add to Pangenome' button. The data will be checked,
submitted and the workflows should kick in!
</p>
</div>


<div id="outline-container-org5d317ac" class="outline-3">
<h3 id="org5d317ac"><span class="section-number-3">5.1</span> Trouble shooting</h3>
<div class="outline-text-3" id="text-5-1">
<p>
Ooops. We got an error saying: {"stem":
"<a href="http://www.wikidata.org/entity/">http://www.wikidata.org/entity/</a>",&#x2026;  which means that our location
field was not formed correctly!  After fixing it to look like
<a href="http://www.wikidata.org/entity/Q65">http://www.wikidata.org/entity/Q65</a> (note http instead on https and
entity instead of wiki) the submission went through. Reload the page
(it won't empty the fields) to re-enable the submit button.
</p>
</div>
</div>
</div>

<div id="outline-container-org389fedb" class="outline-2">
<h2 id="org389fedb"><span class="section-number-2">6</span> Step 4: Check output</h2>
<div class="outline-text-2" id="text-6">
<p>
The current pipeline takes some time to complete! Once it completes
the updated data can be checked on the <a href="./download">DOWNLOAD</a> page. After completion
of above output this <a href="http://sparql.genenetwork.org/sparql/?default-graph-uri=&amp;query=PREFIX+pubseq%3A+%3Chttp%3A%2F%2Fbiohackathon.org%2Fbh20-seq-schema%23MainSchema%2F%3E%0D%0APREFIX+sio%3A+%3Chttp%3A%2F%2Fsemanticscience.org%2Fresource%2F%3E%0D%0Aselect+distinct+%3Fsample+%3Fp+%3Fo%0D%0A%7B%0D%0A+++%3Fsample+sio%3ASIO_000115+%22MT536190.1%22+.%0D%0A+++%3Fsample+%3Fp+%3Fo+.%0D%0A%7D&amp;format=text%2Fhtml&amp;timeout=0&amp;debug=on&amp;run=+Run+Query+">SPARQL query</a> shows some of the metadata we put
in.
</p>
</div>
</div>

<div id="outline-container-org3fb7fc8" class="outline-2">
<h2 id="org3fb7fc8"><span class="section-number-2">7</span> Bulk sequence uploader</h2>
<div class="outline-text-2" id="text-7">
<p>
Above steps require a manual upload of one sequence with metadata.
What if you have a number of sequences you want to upload in bulk?
For this we have a command line version of the uploader that can
directly submit to COVID-19 PubSeq. It accepts a FASTA sequence
file an associated metadata in <a href="https://github.com/arvados/bh20-seq-resource/blob/master/example/maximum_metadata_example.yaml">YAML</a> format. The YAML matches
the web form and gets validated from the same <a href="https://github.com/arvados/bh20-seq-resource/blob/master/bh20sequploader/bh20seq-schema.yml">schema</a> looks. The YAML
that you need to create/generate for your samples looks like
</p>

<p>
A minimal example of metadata looks like
</p>

<div class="org-src-container">
<pre class="src src-json">id: placeholder

license:
    license_type: http://creativecommons.org/licenses/by/4.0/

host:
    host_species: http://purl.obolibrary.org/obo/NCBITaxon_9606

sample:
    sample_id: XX
    collection_date: "2020-01-01"
    collection_location: http://www.wikidata.org/entity/Q148

virus:
    virus_species: http://purl.obolibrary.org/obo/NCBITaxon_2697049

technology:
    sample_sequencing_technology: [http://www.ebi.ac.uk/efo/EFO_0008632]

submitter:
    authors: [John Doe]
</pre>
</div>

<p>
a more elaborate example (note most fields are optional) may look like
</p>

<div class="org-src-container">
<pre class="src src-json">id: placeholder

host:
    host_id: XX1
    host_species: http://purl.obolibrary.org/obo/NCBITaxon_9606
    host_sex: http://purl.obolibrary.org/obo/PATO_0000384
    host_age: 20
    host_age_unit: http://purl.obolibrary.org/obo/UO_0000036
    host_health_status: http://purl.obolibrary.org/obo/NCIT_C25269
    host_treatment: Process in which the act is intended to modify or alter host status (Compounds)
    host_vaccination: [vaccines1,vaccine2]
    ethnicity: http://purl.obolibrary.org/obo/HANCESTRO_0010
    additional_host_information: Optional free text field for additional information

sample:
    sample_id: Id of the sample as defined by the submitter
    collector_name: Name of the person that took the sample
    collecting_institution: Institute that was responsible of sampling
    specimen_source: [http://purl.obolibrary.org/obo/NCIT_C155831,http://purl.obolibrary.org/obo/NCIT_C155835]
    collection_date: "2020-01-01"
    collection_location: http://www.wikidata.org/entity/Q148
    sample_storage_conditions: frozen specimen
    source_database_accession: [http://identifiers.org/insdc/LC522350.1#sequence]
    additional_collection_information: Optional free text field for additional information

virus:
    virus_species: http://purl.obolibrary.org/obo/NCBITaxon_2697049
    virus_strain: SARS-CoV-2/human/CHN/HS_8/2020

technology:
    sample_sequencing_technology: [http://www.ebi.ac.uk/efo/EFO_0009173,http://www.ebi.ac.uk/efo/EFO_0009173]
    alignment_protocol: Protocol used for assembly
    sequencing_coverage: [70.0, 100.0]
    assembly_method: "http://purl.obolibrary.org/obo/GENEPIO_0001628"
    additional_technology_information: Optional free text field for additional information

submitter:
    authors: [John Doe, Joe Boe, Jonny Oe]
    submitter_name: [John Doe]
    submitter_address: John Doe's address
    originating_lab: John Doe kitchen
    lab_address: John Doe's address
    provider: XXX1
    submitter_sample_id: XXX2
    publication: PMID00001113
    submitter_orcid: [https://orcid.org/0000-0000-0000-0000,https://orcid.org/0000-0000-0000-0001]
    additional_submitter_information: Optional free text field for additional information
</pre>
</div>

<p>
more metadata is yummy when stored in RDF. <a href="https://yummydata.org/">Yummydata</a> is useful to a wider community. Note
that many of the terms in above example are URIs, such as
host_species: <a href="http://purl.obolibrary.org/obo/NCBITaxon_9606">http://purl.obolibrary.org/obo/NCBITaxon_9606</a>.  We use
web ontologies for these to make the data less ambiguous and more
FAIR. Check out the option fields as defined in the schema. If it is not listed,
check the <a href="https://github.com/arvados/bh20-seq-resource/blob/master/semantic_enrichment/labels.ttl">labels.ttl</a> file. Also,
a little bit of web searching may be required or <a href="./contact">contact</a> us.
</p>
</div>

<div id="outline-container-org3514567" class="outline-3">
<h3 id="org3514567"><span class="section-number-3">7.1</span> Run the uploader (CLI)</h3>
<div class="outline-text-3" id="text-7-1">
<p>
Installing with pip you should be able to run
</p>

<pre class="example">
bh20sequploader sequence.fasta metadata.yaml
</pre>


<p>
Alternatively the script can be installed from <a href="https://github.com/arvados/bh20-seq-resource#installation">github</a>. Run on the
command line
</p>

<pre class="example">
python3 bh20sequploader/main.py example/sequence.fasta example/maximum_metadata_example.yaml
</pre>


<p>
after installing dependencies (also described in <a href="https://github.com/arvados/bh20-seq-resource/blob/master/doc/INSTALL.md">INSTALL</a> with the GNU
Guix package manager). The <code>--help</code> shows
</p>

<div class="org-src-container">
<pre class="src src-sh">Entering sequence uploader
usage: main.py [-h] [--validate] [--skip-qc] [--trusted] metadata sequence_p1 [sequence_p2]

Upload SARS-CoV-19 sequences for analysis

positional arguments:
  metadata     sequence metadata json
  sequence_p1  sequence FASTA/FASTQ
  sequence_p2  sequence FASTQ pair

optional arguments:
  -h, --help   show this help message and exit
  --validate   Dry run, validate only
  --skip-qc    Skip local qc check
  --trusted    Trust local validation and add directly to validated project
</pre>
</div>

<p>
The web interface using this exact same script so it should just work
(TM).
</p>
</div>
</div>

<div id="outline-container-orgc97a930" class="outline-3">
<h3 id="orgc97a930"><span class="section-number-3">7.2</span> Example: uploading bulk GenBank sequences</h3>
<div class="outline-text-3" id="text-7-2">
<p>
At this point, most of PubSeq's FASTA files come from NCBI <a href="https://www.ncbi.nlm.nih.gov/genbank/">GenBank</a>.
This data is public (see the policy) and we provide it with metadata
under a <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY-4.0</a> license.
</p>

<p>
We use multiple scripts to fetch, check and update data. Since there
are dependencies involved we suggest to match our development work
place and use a Guix environment to run the tools, see also
<a href="../INSTALL.md">INSTALL.md</a>.
</p>

<p>
The scripts to pull data from GenBank are in
<a href="https://github.com/pubseq/bh20-seq-resource/tree/master/workflows/pull-data/genbank">workflows/pull-data/genbank</a>. The scripts that query Pubseq are in
<a href="https://github.com/pubseq/bh20-seq-resource/tree/master/workflows/pubseq">workflows/pubseq</a>.
</p>
</div>

<div id="outline-container-orga76159c" class="outline-4">
<h4 id="orga76159c"><span class="section-number-4">7.2.1</span> List PubSeq IDs</h4>
<div class="outline-text-4" id="text-7-2-1">
<p>
The first script to run fetches a <a href="https://github.com/pubseq/bh20-seq-resource/blob/master/workflows/pubseq/pubseq-fetch-ids">list of Pubseq IDs</a> to make sure we
don't download data already in PubSeq. It requires Ruby3 and
nothing else as a dependency and takes a second to run:
</p>

<div class="org-src-container">
<pre class="src src-sh">ruby pubseq-fetch-ids &gt; pubseq_ids.txt
head pubseq_ids.txt
  MT459889.1
  MT509498.1
  SRR11622145
  ERR4304866
  SRR11916007
</pre>
</div>

<p>
showing all GenBank IDs stored in PubSeq.
</p>
</div>
</div>

<div id="outline-container-org89757c4" class="outline-4">
<h4 id="org89757c4"><span class="section-number-4">7.2.2</span> Fetch GenBank IDs</h4>
<div class="outline-text-4" id="text-7-2-2">
<p>
In the next step we essentially follow the PubSeq <a href="https://github.com/pubseq/bh20-seq-resource/tree/master/workflows/pull-data/genbank">GenBank README</a> and
fetch the GenBank IDs as a list with
</p>

<div class="org-src-container">
<pre class="src src-sh">python3 genbank-fetch-ids.py --skip pubseq_ids.txt &gt; genbank_ids.txt
</pre>
</div>

<p>
this list is what tells us to download from GenBank with the next
script
</p>
</div>
</div>

<div id="outline-container-org538bf18" class="outline-4">
<h4 id="org538bf18"><span class="section-number-4">7.2.3</span> Fetch GenBank XML</h4>
<div class="outline-text-4" id="text-7-2-3">
<p>
Here we fetch the XML files for all the IDs that are listed in
genbank_ids.txt. This is a slow procedure!
</p>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #b0bec5;"># </span><span style="color: #b0bec5;">--- fetch XML</span>
python3 update-from-genbank.py --ids genbank_ids.txt --out ~/tmp/genbank
</pre>
</div>

<p>
Sometimes the download stops. In that case you can restart the
download with above command. It will only fetch the missing files.
With the same <code>genbank_ids.txt</code> file that should work fine.
</p>
</div>
</div>

<div id="outline-container-orgf96d0f6" class="outline-4">
<h4 id="orgf96d0f6"><span class="section-number-4">7.2.4</span> Transform and normalize data for uploading</h4>
<div class="outline-text-4" id="text-7-2-4">
<p>
Now we have the GenBank XML data we can start transforming
and the metadata
</p>

<div class="org-src-container">
<pre class="src src-sh"><span style="color: #b0bec5;"># </span><span style="color: #b0bec5;">--- Basic transform to YAML/JSON and FASTA</span>
python3 transform-genbank-xml2yamlfa.py --out ~/tmp/pubseq [XML file(s)]
</pre>
</div>

<p>
which also writes a file named 'state.json' in the output directory.
This file contains all errors and warnings! For example we find
</p>

<div class="org-src-container">
<pre class="src src-js"><span style="color: #9ccc65;">"MT665288"</span>: {
    <span style="color: #9ccc65;">"valid"</span>: <span style="color: #8bc34a;">false</span>,
    <span style="color: #9ccc65;">"error"</span>: <span style="color: #9ccc65;">"Sequence too short for MT665288"</span>,
    <span style="color: #9ccc65;">"warnings"</span>: [
        <span style="color: #9ccc65;">"Missing host_species"</span>,
        <span style="color: #9ccc65;">"Missing collection_location"</span>,
        <span style="color: #9ccc65;">"Missing collection_date"</span>,
        <span style="color: #9ccc65;">"Missing host_species"</span>,
        <span style="color: #9ccc65;">"Missing specimen_source"</span>
    ]
},
</pre>
</div>

<p>
Ouch! Not only does it fail the sequence, it also fails on other
metadata fields. This will mean this record gets dropped.
</p>

<p>
Note we split transformation from normalizing metadata. Mostly
because transformation is related to the source of the data. In this case
we transform GenBank XML to first stage JSON.
</p>
</div>
</div>

<div id="outline-container-org274ef05" class="outline-4">
<h4 id="org274ef05"><span class="section-number-4">7.2.5</span> Normalize metadata</h4>
<div class="outline-text-4" id="text-7-2-5">
<p>
In the next stage we adjust and normalize the metadata so it can be
transformed to RDF. This transformation checks data and transforms
ambiguous statements into 'absolute' statements where possible.  For
example human and Homo sapiens mean the same thing and translate to
the unambiguous URI <a href="http://purl.obolibrary.org/obo/NCBITaxon_9606">http://purl.obolibrary.org/obo/NCBITaxon_9606</a>.
</p>

<p>
Note that, in addition to the earlier state.json file, which refers to
the input YAML/JSON files, we pass in two optional comma separated
transformation files which are simple mappings, e.g. for above
</p>

<pre class="example">
Homo sapiens,http://purl.obolibrary.org/obo/NCBITaxon_9606
</pre>


<p>
Do note, however, that we are increasingly moving to handling such
mappings in regex <a href="https://github.com/pubseq/bh20-seq-resource/blob/master/workflows/pubseq/normalize/mapping.py">code</a>.
</p>

<p>
To run the normalization use the command:
</p>

<div class="org-src-container">
<pre class="src src-sh">python3 normalize-step1.py -s ~/tmp/yamlfa/state.json --species ncbi_host_species.csv --specimen specimen.csv --validate
</pre>
</div>

<p>
To work on one single JSON/YAML file/record, you can pass in the ID,
e.g.
</p>

<div class="org-src-container">
<pre class="src src-sh">python3 normalize-step1.py -s ~/tmp/new-yamlfa-orig.2/state.json --validate MW084447
</pre>
</div>

<p>
To write output specify the output dir:
</p>

<div class="org-src-container">
<pre class="src src-sh">python3 normalize-step1.py -s ~/tmp/new-yamlfa-orig.2/state.json --validate MW084447 --out ~/tmp/test
</pre>
</div>

<p>
and run a diff with, for example,
</p>

<div class="org-src-container">
<pre class="src src-sh">colordiff ~/tmp/new-yamlfa-orig.2/MW084447.json ~/tmp/test/MW084447.json
</pre>
</div>

<p>
or a more fancy version using the excellent <a href="https://stedolan.github.io/jq/">jq tool</a>:
</p>

<div class="org-src-container">
<pre class="src src-sh">colordiff &lt;(jq -S . ~/tmp/new-yamlfa-orig.2/MW084447.json) &lt;(jq -S . ~/tmp/test/MW084447.json )
</pre>
</div>

<p>
which shows the transformation very clearly:
</p>

<div class="org-src-container">
<pre class="src src-diff"><span style="color: #b39ddb;">3c3</span>
<span style="color: #aa2222;">&lt;</span><span style="color: #ff9800;">     "host_species": "</span><span style="color: #ff9800;">Homo sapiens</span><span style="color: #ff9800;">"</span>
<span style="color: #b39ddb;">---</span>
&gt;     "host_species": "<span style="color: #8bc34a;">http://purl.obolibrary.org/obo/NCBITaxon_9606</span>"
</pre>
</div>
</div>
</div>

<div id="outline-container-org2351e31" class="outline-4">
<h4 id="org2351e31"><span class="section-number-4">7.2.6</span> Upload data</h4>
</div>
</div>

<div id="outline-container-org5985f41" class="outline-3">
<h3 id="org5985f41"><span class="section-number-3">7.3</span> Example: preparing metadata from spreadsheets</h3>
<div class="outline-text-3" id="text-7-3">
<p>
Usually, metadata are available in a tabular format, such as
spreadsheets (the GenBank data entry has that as a default). As an
example, we provide a script <a href="https://github.com/arvados/bh20-seq-resource/tree/master/scripts/esr_samples">esr_samples.py</a> to show you how to parse
your metadata from a spreadsheet into YAML files, from a template,
ready for PubSeq upload. To execute the script, go in the
~bh20-seq-resource/scripts/esr_samples and execute
</p>

<div class="org-src-container">
<pre class="src src-sh">python3 esr_samples.py
</pre>
</div>

<p>
You will find the YAML files in the `yaml` folder which will be
created in the same directory.
</p>

<p>
In the example we use Python pandas to read the spreadsheet into a
tabular structure. Next we use a <a href="https://github.com/arvados/bh20-seq-resource/blob/master/scripts/esr_samples/template.yaml">template.yaml</a> file that gets filled
in by <code>esr_samples.py</code> so we get a metadata YAML file for each sample.
</p>

<p>
Next run the earlier CLI uploader for each YAML and FASTA combination.
It can't be much easier than this. For ESR we uploaded a batch of 600
sequences this way writing a few lines of Python <a href="https://github.com/arvados/bh20-seq-resource/blob/master/scripts/esr_samples/esr_samples.py">code</a>. See <a href="http://covid19.genenetwork.org/resource/20VR0995">example</a>.
</p>
</div>
</div>
</div>
</div>
<div id="postamble" class="status">
<hr><small>Created by <a href="http://thebird.nl/">Pjotr Prins</a> (pjotr.public768 at thebird 'dot' nl) using Emacs org-mode and a healthy dose of Lisp!<br />Modified 2021-05-09 Sun 02:52</small>.
</div>
</body>
</html>
