<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<!-- 2021-04-23 Fri 09:31 -->
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>About/FAQ</title>
<meta name="generator" content="Org mode" />
<meta name="author" content="Pjotr Prins" />
<style type="text/css">
 <!--/*--><![CDATA[/*><!--*/
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #ccc;
    box-shadow: 3px 3px 3px #eee;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: auto;
    padding-top: 1.2em;
  }
  pre.src:before {
    display: none;
    position: absolute;
    background-color: white;
    top: -10px;
    right: 10px;
    padding: 3px;
    border: 1px solid black;
  }
  pre.src:hover:before { display: inline;}
  /* Languages per Org manual */
  pre.src-asymptote:before { content: 'Asymptote'; }
  pre.src-awk:before { content: 'Awk'; }
  pre.src-C:before { content: 'C'; }
  /* pre.src-C++ doesn't work in CSS */
  pre.src-clojure:before { content: 'Clojure'; }
  pre.src-css:before { content: 'CSS'; }
  pre.src-D:before { content: 'D'; }
  pre.src-ditaa:before { content: 'ditaa'; }
  pre.src-dot:before { content: 'Graphviz'; }
  pre.src-calc:before { content: 'Emacs Calc'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-fortran:before { content: 'Fortran'; }
  pre.src-gnuplot:before { content: 'gnuplot'; }
  pre.src-haskell:before { content: 'Haskell'; }
  pre.src-hledger:before { content: 'hledger'; }
  pre.src-java:before { content: 'Java'; }
  pre.src-js:before { content: 'Javascript'; }
  pre.src-latex:before { content: 'LaTeX'; }
  pre.src-ledger:before { content: 'Ledger'; }
  pre.src-lisp:before { content: 'Lisp'; }
  pre.src-lilypond:before { content: 'Lilypond'; }
  pre.src-lua:before { content: 'Lua'; }
  pre.src-matlab:before { content: 'MATLAB'; }
  pre.src-mscgen:before { content: 'Mscgen'; }
  pre.src-ocaml:before { content: 'Objective Caml'; }
  pre.src-octave:before { content: 'Octave'; }
  pre.src-org:before { content: 'Org mode'; }
  pre.src-oz:before { content: 'OZ'; }
  pre.src-plantuml:before { content: 'Plantuml'; }
  pre.src-processing:before { content: 'Processing.js'; }
  pre.src-python:before { content: 'Python'; }
  pre.src-R:before { content: 'R'; }
  pre.src-ruby:before { content: 'Ruby'; }
  pre.src-sass:before { content: 'Sass'; }
  pre.src-scheme:before { content: 'Scheme'; }
  pre.src-screen:before { content: 'Gnu Screen'; }
  pre.src-sed:before { content: 'Sed'; }
  pre.src-sh:before { content: 'shell'; }
  pre.src-sql:before { content: 'SQL'; }
  pre.src-sqlite:before { content: 'SQLite'; }
  /* additional languages in org.el's org-babel-load-languages alist */
  pre.src-forth:before { content: 'Forth'; }
  pre.src-io:before { content: 'IO'; }
  pre.src-J:before { content: 'J'; }
  pre.src-makefile:before { content: 'Makefile'; }
  pre.src-maxima:before { content: 'Maxima'; }
  pre.src-perl:before { content: 'Perl'; }
  pre.src-picolisp:before { content: 'Pico Lisp'; }
  pre.src-scala:before { content: 'Scala'; }
  pre.src-shell:before { content: 'Shell Script'; }
  pre.src-ebnf2ps:before { content: 'ebfn2ps'; }
  /* additional language identifiers per "defun org-babel-execute"
       in ob-*.el */
  pre.src-cpp:before  { content: 'C++'; }
  pre.src-abc:before  { content: 'ABC'; }
  pre.src-coq:before  { content: 'Coq'; }
  pre.src-groovy:before  { content: 'Groovy'; }
  /* additional language identifiers from org-babel-shell-names in
     ob-shell.el: ob-shell is the only babel language using a lambda to put
     the execution function name together. */
  pre.src-bash:before  { content: 'bash'; }
  pre.src-csh:before  { content: 'csh'; }
  pre.src-ash:before  { content: 'ash'; }
  pre.src-dash:before  { content: 'dash'; }
  pre.src-ksh:before  { content: 'ksh'; }
  pre.src-mksh:before  { content: 'mksh'; }
  pre.src-posh:before  { content: 'posh'; }
  /* Additional Emacs modes also supported by the LaTeX listings package */
  pre.src-ada:before { content: 'Ada'; }
  pre.src-asm:before { content: 'Assembler'; }
  pre.src-caml:before { content: 'Caml'; }
  pre.src-delphi:before { content: 'Delphi'; }
  pre.src-html:before { content: 'HTML'; }
  pre.src-idl:before { content: 'IDL'; }
  pre.src-mercury:before { content: 'Mercury'; }
  pre.src-metapost:before { content: 'MetaPost'; }
  pre.src-modula-2:before { content: 'Modula-2'; }
  pre.src-pascal:before { content: 'Pascal'; }
  pre.src-ps:before { content: 'PostScript'; }
  pre.src-prolog:before { content: 'Prolog'; }
  pre.src-simula:before { content: 'Simula'; }
  pre.src-tcl:before { content: 'tcl'; }
  pre.src-tex:before { content: 'TeX'; }
  pre.src-plain-tex:before { content: 'Plain TeX'; }
  pre.src-verilog:before { content: 'Verilog'; }
  pre.src-vhdl:before { content: 'VHDL'; }
  pre.src-xml:before { content: 'XML'; }
  pre.src-nxml:before { content: 'XML'; }
  /* add a generic configuration mode; LaTeX export needs an additional
     (add-to-list 'org-latex-listings-langs '(conf " ")) in .emacs */
  pre.src-conf:before { content: 'Configuration File'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .equation-container {
    display: table;
    text-align: center;
    width: 100%;
  }
  .equation {
    vertical-align: middle;
  }
  .equation-label {
    display: table-cell;
    text-align: right;
    vertical-align: middle;
  }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  .org-svg { width: 90%; }
  /*]]>*/-->
</style>
<script type="text/javascript">
// @license magnet:?xt=urn:btih:e95b018ef3580986a04669f1b5879592219e2a7a&dn=public-domain.txt Public Domain
<!--/*--><![CDATA[/*><!--*/
     function CodeHighlightOn(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.add("code-highlighted");
         target.classList.add("code-highlighted");
       }
     }
     function CodeHighlightOff(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.remove("code-highlighted");
         target.classList.remove("code-highlighted");
       }
     }
    /*]]>*///-->
// @license-end
</script>
</head>
<body>
<div id="content">
<h1 class="title">About/FAQ</h1>
<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#orgc09f111">1. What is the 'public sequence resource' about?</a></li>
<li><a href="#org0d01448">2. Presentations</a></li>
<li><a href="#orgbbe5a7c">3. Who created the public sequence resource?</a></li>
<li><a href="#org003d046">4. How does the public sequence resource compare to other data resources?</a></li>
<li><a href="#org05f2eb2">5. Why should I upload my data here?</a></li>
<li><a href="#org5a56994">6. Why should I not upload by data here?</a></li>
<li><a href="#orgefc889a">7. How does the public sequence resource work?</a></li>
<li><a href="#org372cc70">8. Who uses the public sequence resource?</a></li>
<li><a href="#org34940b2">9. How can I contribute?</a></li>
<li><a href="#org1f715bd">10. Is this about open data?</a></li>
<li><a href="#org09d0632">11. Is this about free software?</a></li>
<li><a href="#orgf31be70">12. How do I upload raw data?</a></li>
<li><a href="#org5011320">13. How do I change metadata?</a></li>
<li><a href="#org94061db">14. How do I change the work flows?</a></li>
<li><a href="#org4ccc897">15. How do I change the source code?</a></li>
<li><a href="#org45ba9a6">16. Should I choose CC-BY or CC0?</a></li>
<li><a href="#org0f8f0c4">17. Are there also variant in the RDF databases?</a></li>
<li><a href="#org3912e77">18. How do I deal with private data and privacy?</a></li>
<li><a href="#org1473e4b">19. Do you have any checks or concerns if human sequence accidentally submitted to your service as part of a fastq? *</a></li>
<li><a href="#org51420ae">20. Does PubSeq support only SARS-CoV-2 data?</a></li>
<li><a href="#org8f0ab64">21. Contact: how do I communicate with you?</a></li>
<li><a href="#org556b9fd">22. Citing PubSeq</a></li>
<li><a href="#org0603090">23. Who are the sponsors?</a></li>
</ul>
</div>
</div>

<div id="outline-container-orgc09f111" class="outline-2">
<h2 id="orgc09f111"><span class="section-number-2">1</span> What is the 'public sequence resource' about?</h2>
<div class="outline-text-2" id="text-1">
<p>
PubSeq, the <b>public sequence resource</b>, aims to provide a generic and
useful resource for COVID-19 research.  The focus is on providing the
best possible sequence data with associated metadata that can be used
for sequence comparison and protein prediction.
</p>

<p>
Just to avoid the notion that PubSeq is a biorepository: it is
different. We are not competing with the likes of GenBank and EBI/ENA
Open Data - in fact, PubSeq federates open data and adds useful
normalised metadata and opens up running workflows by anyone on that
data.
</p>

<p>
Note that there is no conflict of posting data to multiple
repositories.
</p>

<p>
It is better to view PubSeq as an <b>open precision medicine initiative</b> that
allows for early detection of new variants and strains; very valuable
for both testing and treatment. We have live analysis work flows that
run on uploading a new sequence. Contributing sequences helps the
overall detection. At this moment the output is limited.  We will soon
add interactive discovery tools.  Uploading data from a sequencer will
give researchers all information they want in 5 hours <b>without any
bioinformatics knowledge</b>. We may even provide workflows that take
data straight from the sequencer.  In fact, you get very little out of
existing biorepositories, in our opinion, unless you have solid
bioinformatics support. And even then there is the question about
comparing data that was created using different technologies and
workflows. How do you deal with that?
</p>

<p>
With PubSeq we are in a position to reanalyse that data from raw
material when new insights arise.  It is likely that COVID19 will be
around for another year at least.  After that we should prepare for
the next pandemic. PubSeq is meant for that. We take the long view and
it will only get more powerful.  Who knows: maybe the live analysis
part of PubSeq will be an approach that other biorepositories will
follow.
</p>
</div>
</div>

<div id="outline-container-org0d01448" class="outline-2">
<h2 id="org0d01448"><span class="section-number-2">2</span> Presentations</h2>
<div class="outline-text-2" id="text-2">
<p>
We presented at the BOSC 2020 Have a look at the <a href="https://bcc2020.sched.com/event/coLw">video</a> (<a href="https://drive.google.com/file/d/1skXHwVKM_gl73-_4giYIOQ1IlC5X5uBo/view?usp=sharing">alternative
link</a>) and the <a href="https://drive.google.com/file/d/1vyEgfvSqhM9yIwWZ6Iys-QxhxtVxPSdp/view?usp=sharing">poster</a>.
</p>
</div>
</div>

<div id="outline-container-orgbbe5a7c" class="outline-2">
<h2 id="orgbbe5a7c"><span class="section-number-2">3</span> Who created the public sequence resource?</h2>
<div class="outline-text-2" id="text-3">
<p>
The <b>public sequence resource</b> is an initiative by <a href="https://github.com/arvados/bh20-seq-resource/graphs/contributors">bioinformatics</a> and
ontology experts who want to create something agile and useful for the
wider research community. The initiative started at the COVID-19
biohackathon in April 2020 and is ongoing. The main project drivers
are Pjotr Prins (UTHSC), Peter Amstutz (Curii), Andrea Guarracino
(University of Rome Tor Vergata), Michael Crusoe (Common Workflow
Language), Thomas Liener (consultant, formerly EBI), Erik Garrison
(UCSC) and Jerven Bolleman (Swiss Institute of Bioinformatics).
</p>

<p>
Notably, as this is a free software initiative, the project represents
major work by hundreds of software developers and ontology and data
wrangling experts. Thank you everyone!
</p>
</div>
</div>

<div id="outline-container-org003d046" class="outline-2">
<h2 id="org003d046"><span class="section-number-2">4</span> How does the public sequence resource compare to other data resources?</h2>
<div class="outline-text-2" id="text-4">
<p>
The short version is that we use state-of-the-art practices in
bioinformatics using agile methods. Unlike the resources from large
institutes we can improve things on a dime and anyone can contribute
to building out this resource! Sequences from GenBank, EBI/ENA and
others are regularly added to PubSeq. We encourage people to everyone
to submit on PubSeq because of its superior live tooling and metadata
support (see the next question).
</p>

<p>
Importantly: all data is published under either the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons
4.0 attribution license</a> or the <a href="https://creativecommons.org/share-your-work/public-domain/cc0/">CC0 “No Rights Reserved” license</a> which
means it data can be published and workflows can run in public
environments allowing for improved access for research and
reproducible results. This contrasts with some other public resources,
such as GISAID.
</p>
</div>
</div>

<div id="outline-container-org05f2eb2" class="outline-2">
<h2 id="org05f2eb2"><span class="section-number-2">5</span> Why should I upload my data here?</h2>
<div class="outline-text-2" id="text-5">
<ol class="org-ol">
<li>We champion truly shareable data without licensing restrictions - with proper
attribution</li>
<li>We provide full metadata support using state-of-the-art ontology's</li>
<li>We provide a web-based sequence uploader and a command-line version
for bulk uploads</li>
<li>We provide a live SPARQL end-point for all metadata</li>
<li>We provide free data analysis and sequence comparison triggered on data upload</li>
<li>We do real work for you, with this <a href="https://workbench.lugli.arvadosapi.com/container_requests/lugli-xvhdp-bhhk4nxx1lch5od">link</a> you can see the last
run took 5.5 hours!</li>
<li>We provide free downloads of all computed output</li>
<li>There is no need to set up pipelines and/or compute clusters</li>
<li>All workflows get triggered on uploading a new sequence</li>
<li>When someone (you?) improves the software/workflows and everyone benefits</li>
<li>Your data gets automatically integrated with the Swiss Institure of
Bioinformatics COVID-19 knowledge base
<a href="https://covid-19-sparql.expasy.org/">https://covid-19-sparql.expasy.org/</a> (Elixir Switzerland)</li>
<li>Your data will be used to develop drug targets</li>
</ol>

<p>
Finally, if you upload your data here we have workflows that output
formatted data suitable for <a href="http://covid19.genenetwork.org/blog?id=using-covid-19-pubseq-part6">uploading to EBI resources</a> (and soon
others). Uploading your data here get your data ready for upload to
multiple resources.
</p>
</div>
</div>

<div id="outline-container-org5a56994" class="outline-2">
<h2 id="org5a56994"><span class="section-number-2">6</span> Why should I not upload by data here?</h2>
<div class="outline-text-2" id="text-6">
<p>
Funny question.  There are only good reasons to upload your data here
and make it available to the widest audience possible.
</p>

<p>
In fact, you can upload your data here as well as to other
resources. It is your data after all. No one can prevent you from
uploading your data to multiple resources.
</p>

<p>
We recommend uploading to EBI and NCBI resources using our data
conversion tools. It means you only enter data once and make the
process smooth. You can also use our command line data uploader
for bulk uploads!
</p>
</div>
</div>

<div id="outline-container-orgefc889a" class="outline-2">
<h2 id="orgefc889a"><span class="section-number-2">7</span> How does the public sequence resource work?</h2>
<div class="outline-text-2" id="text-7">
<p>
On uploading a sequence with metadata it will automatically be
processed and incorporated into the public pangenome with metadata
using workflows from the High Performance Open Biology Lab defined
<a href="https://github.com/hpobio-lab/viral-analysis/tree/master/cwl/pangenome-generate">here</a>.
</p>
</div>
</div>

<div id="outline-container-org372cc70" class="outline-2">
<h2 id="org372cc70"><span class="section-number-2">8</span> Who uses the public sequence resource?</h2>
<div class="outline-text-2" id="text-8">
<p>
The Swiss Institute of Bioinformatics has included this data in the
SPARQL <a href="https://covid-19-sparql.expasy.org/">https://covid-19-sparql.expasy.org/</a> endpoint.
</p>

<p>
The Pantograph <a href="https://graph-genome.github.io/">viewer</a> uses PubSeq data for their visualisations.
</p>

<p>
<a href="https://uthsc.edu">UTHSC</a> (USA), <a href="https://www.esr.cri.nz/">ESR</a> (New Zealand) and <a href="https://www.ornl.gov/news/ornl-fight-against-covid-19">ORNL</a> (USA) use COVID-19 PubSeq data
for monitoring, protein prediction and drug development.
</p>

<p>
Amazon AWS has included PubSeq in the <a href="https://registry.opendata.aws/pubseq/">Amazon AWS OpenData</a> registry.
</p>
</div>
</div>

<div id="outline-container-org34940b2" class="outline-2">
<h2 id="org34940b2"><span class="section-number-2">9</span> How can I contribute?</h2>
<div class="outline-text-2" id="text-9">
<p>
You can contribute by submitting sequences, updating metadata, submit
issues on our issue tracker, and more importantly add functionality.
See 'How do I change the source code' below. Read through our online
documentation at <a href="http://covid19.genenetwork.org/blog">http://covid19.genenetwork.org/blog</a> as a starting
point.
</p>
</div>
</div>

<div id="outline-container-org1f715bd" class="outline-2">
<h2 id="org1f715bd"><span class="section-number-2">10</span> Is this about open data?</h2>
<div class="outline-text-2" id="text-10">
<p>
All data is published under a <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons 4.0 attribution license</a>
(CC-BY-4.0). You can download the raw and published (GFA/RDF/FASTA)
data and store it for further processing.
</p>
</div>
</div>

<div id="outline-container-org09d0632" class="outline-2">
<h2 id="org09d0632"><span class="section-number-2">11</span> Is this about free software?</h2>
<div class="outline-text-2" id="text-11">
<p>
Absolutely. Free software allows for fully reproducible pipelines. You
can take our workflows and data and run it elsewhere!
</p>
</div>
</div>

<div id="outline-container-orgf31be70" class="outline-2">
<h2 id="orgf31be70"><span class="section-number-2">12</span> How do I upload raw data?</h2>
<div class="outline-text-2" id="text-12">
<p>
We are preparing raw sequence data pipelines (fastq and BAM). The
reason is that we want the best data possible for downstream analysis
(including protein prediction and test development). The current
approach where people publish final sequences of SARS-CoV-2 is lacking
because it hides how this sequence was created. For reasons of
reproducible and improved results we want/need to work with the raw
sequence reads (both short reads and long reads) and take alternative
assembly variations into consideration. This is all work in progress.
</p>
</div>
</div>

<div id="outline-container-org5011320" class="outline-2">
<h2 id="org5011320"><span class="section-number-2">13</span> How do I change metadata?</h2>
<div class="outline-text-2" id="text-13">
<p>
See the <a href="http://covid19.genenetwork.org/blog">http://covid19.genenetwork.org/blog</a>!
</p>
</div>
</div>

<div id="outline-container-org94061db" class="outline-2">
<h2 id="org94061db"><span class="section-number-2">14</span> How do I change the work flows?</h2>
<div class="outline-text-2" id="text-14">
<p>
Workflows are on <a href="https://github.com/arvados/bh20-seq-resource/tree/master/workflows">github</a> and can be modified. See also the BLOG
<a href="http://covid19.genenetwork.org/blog">http://covid19.genenetwork.org/blog</a> on workflows.
</p>
</div>
</div>

<div id="outline-container-org4ccc897" class="outline-2">
<h2 id="org4ccc897"><span class="section-number-2">15</span> How do I change the source code?</h2>
<div class="outline-text-2" id="text-15">
<p>
Go to our <a href="https://github.com/arvados/bh20-seq-resource">source code repositories</a>, fork/clone the repository, change
something and submit a <a href="https://github.com/arvados/bh20-seq-resource/pulls">pull request</a> (PR). That easy! Check out how
many PRs we already merged.
</p>
</div>
</div>

<div id="outline-container-org45ba9a6" class="outline-2">
<h2 id="org45ba9a6"><span class="section-number-2">16</span> Should I choose CC-BY or CC0?</h2>
<div class="outline-text-2" id="text-16">
<p>
Restrictive data licenses are hampering data sharing and reproducible
research. CC0 is the preferred license because it gives researchers
the most freedom. Since we provide metadata there is no reason for
others not to honour your work. We also provide CC-BY as an option
because we know people like the attribution clause.
</p>

<p>
In all honesty: we prefer both data and software to be free.
</p>
</div>
</div>

<div id="outline-container-org0f8f0c4" class="outline-2">
<h2 id="org0f8f0c4"><span class="section-number-2">17</span> Are there also variant in the RDF databases?</h2>
<div class="outline-text-2" id="text-17">
<p>
We do output a RDF file with the pangenome built in, and you can parse it because it has variants implicitly.
</p>

<p>
We are also writing tools to generate VCF files directly from the pangenome.
</p>
</div>
</div>

<div id="outline-container-org3912e77" class="outline-2">
<h2 id="org3912e77"><span class="section-number-2">18</span> How do I deal with private data and privacy?</h2>
<div class="outline-text-2" id="text-18">
<p>
A public sequence resource is about public data. Metadata can refer to
private data. You can use your own (anonymous) identifiers.  We also
plan to combine identifiers with clinical data stored securely at
<a href="https://redcap-covid19.elixir-luxembourg.org/redcap/">REDCap</a>. See the relevant <a href="https://github.com/arvados/bh20-seq-resource/issues/21">tracker</a> for more information and contributing.
</p>
</div>
</div>

<div id="outline-container-org1473e4b" class="outline-2">
<h2 id="org1473e4b"><span class="section-number-2">19</span> Do you have any checks or concerns if human sequence accidentally submitted to your service as part of a fastq? *</h2>
<div class="outline-text-2" id="text-19">
<p>
We are planning to remove reads that match the human reference.
</p>
</div>
</div>

<div id="outline-container-org51420ae" class="outline-2">
<h2 id="org51420ae"><span class="section-number-2">20</span> Does PubSeq support only SARS-CoV-2 data?</h2>
<div class="outline-text-2" id="text-20">
<p>
To date, PubSeq is a resource specific to SARS-CoV-2, but we are designing it to be able to support other species in the future.
</p>
</div>
</div>


<div id="outline-container-org8f0ab64" class="outline-2">
<h2 id="org8f0ab64"><span class="section-number-2">21</span> Contact: how do I communicate with you?</h2>
<div class="outline-text-2" id="text-21">
<p>
We use a <a href="https://gitter.im/arvados/pubseq?utm_source=share-link&amp;utm_medium=link&amp;utm_campaign=share-link">gitter channel</a> you can join. See also <a href="./contact">contact</a>.
</p>
</div>
</div>

<div id="outline-container-org556b9fd" class="outline-2">
<h2 id="org556b9fd"><span class="section-number-2">22</span> Citing PubSeq</h2>
<div class="outline-text-2" id="text-22">
<p>
We have two publications in the works. Until we have a DOI please cite
PubSeq in the following way:
</p>

<p>
We made use of the COVID-19 public sequence (PubSeq) resources hosted
at <a href="http://covid19.genenetwork.org/">http://covid19.genenetwork.org/</a>.
</p>
</div>
</div>


<div id="outline-container-org0603090" class="outline-2">
<h2 id="org0603090"><span class="section-number-2">23</span> Who are the sponsors?</h2>
<div class="outline-text-2" id="text-23">
<p>
The main sponsors are listed in the footer. In addition to the time
generously donated by many contributors we also acknowledge Amazon AWS
for donating COVID-19 related compute time.
</p>
</div>
</div>
</div>
<div id="postamble" class="status">
<hr><small>Created by <a href="http://thebird.nl/">Pjotr Prins</a> (pjotr.public768 at thebird 'dot' nl) using Emacs org-mode and a healthy dose of Lisp!<br />Modified 2021-04-23 Fri 09:26</small>.
</div>
</body>
</html>
